chia/consensus/blockchain.py

```diff
@@ -290,6 +291,7 @@ async def add_block(
         self,
         block: FullBlock,
         pre_validation_result: PreValidationResult,
+        bls_cache: Optional[BLSCache],
```

Comment:
It seems odd to have this here as I'd expect both this and `full_node` to be long-lived objects. Would it be possible to just pass in the cache whenever it's needed?